filesdk/index.js

```diff
@@ -78,15 +78,19 @@ let extractZip = (source, destination) => {
     return extractWithUnzipStream(source, destination);
 }
 let extractWithUnzipStream = (src, dest) => {
-	let defer = q.defer();
-
-	fs.createReadStream(src)
+        let defer = q.defer();
+        fs.createReadStream(src)
 		.pipe(unzip.Extract({ path: dest }))
 		.on('close', (err) => {
-			if (err) defer.reject(err);
-			else defer.resolve('Done!!!');
-        });
-
+			if (err) {
+			    defer.reject(err);
+			}	
+			else {
+                defer.resolve('Done!!!');
+			}	
+        	}).on('error', (err) => {
+			    defer.reject(err);
```

Comment:
Since the extraction of ecar files is happening sequentially now, have we verified that this doesn't cause the extraction of the remaining ecar files to fail if one of them is corrupt?

Reply:
not handled here

Reply:
not handled here, must be done where they are extracted as a list